code/go/0chain.net/smartcontract/storagesc/stakepool.go

```diff
@@ -115,6 +117,32 @@ func (sp *stakePool) cleanStake() (stake currency.Coin, err error) {
 	return staked - sp.TotalUnStake, nil
 }
 
+func (sp *stakePool) stakeForProvider(providerType spenum.Provider, providerID string, balances chainstate.StateContextI) error {
+	staked, err := sp.stake()
+	if err != nil {
+		return err
+	}
+
+	logging.Logger.Info("emitting stake event")
+	data := dbs.DbUpdates{
+		Id:      providerID,
+		Updates: map[string]interface{}{},
+	}
+
+	switch providerType {
+	case spenum.Blobber:
+		tag, data := event.NewUpdateBlobberTotalStakeEvent(providerID, staked)
+		balances.EmitEvent(event.TypeStats, tag, providerID, data)
+	case spenum.Validator:
+		data.Updates["stake_total"] = int64(staked)
```

Comment:
update it with entity, no need to use named field

Reply:
Do you mean updating with struct instead of map?